src/handlers/next/create-app-register-handler.test.ts

```diff
@@ -90,4 +93,160 @@ describe("create-app-register-handler", () => {
+        jwks: mockJwksValue,
+        appId: mockAppId,
+      };
+
+      await handler(req, res);
+
+      expect(mockOnRequestStart).toHaveBeenCalledWith(
+        expect.anything(/* Assume original request */),
+        expect.objectContaining({
+          authToken: "mock-auth-token",
+          saleorDomain: "https://mock-saleor-domain.saleor.cloud/",
+          saleorApiUrl: "https://mock-saleor-domain.saleor.cloud/graphql/",
+        })
+      );
+      expect(mockOnRequestVerified).toHaveBeenCalledWith(
+        expect.anything(/* Assume original request */),
+        expect.objectContaining({
+          authData: expectedAuthData,
+        })
+      );
+      expect(mockOnAuthAplSaved).toHaveBeenCalledWith(
+        expect.anything(/* Assume original request */),
+        expect.objectContaining({
+          authData: expectedAuthData,
+        })
+      );
+      expect(mockOnAuthAplFailed).not.toHaveBeenCalled();
+    });
+
+    it("Runs callback hooks - failed saving to APL scenario", async () => {
+      const mockOnAuthAplFailed = vi.fn();
+      const mockOnAuthAplSaved = vi.fn();
+
+      (mockApl.set as Mock).mockImplementationOnce(() => {
+        throw new Error("test error");
+      });
+
+      const { res, req } = createMocks({
+        /**
+         * Use body, instead of params, otherwise - for some reason - param is not accessible in mock request
+         * Maybe this is a bug https://github.com/howardabrams/node-mocks-http/blob/master/lib/mockRequest.js
+         */
+        body: {
+          auth_token: "mock-auth-token",
+        },
+        headers: {
+          host: "some-saleor-host.cloud",
+          "x-forwarded-proto": "https",
+          "saleor-api-url": "https://mock-saleor-domain.saleor.cloud/graphql/",
+          "saleor-domain": "https://mock-saleor-domain.saleor.cloud/",
+        },
+        method: "POST",
+      });
+
+      const handler = createAppRegisterHandler({
+        apl: mockApl,
+        onAuthAplFailed: mockOnAuthAplFailed,
+        onAuthAplSaved: mockOnAuthAplSaved,
+      });
+
+      const expectedAuthData: AuthData = {
+        token: "mock-auth-token",
+        domain: "https://mock-saleor-domain.saleor.cloud/",
+        saleorApiUrl: "https://mock-saleor-domain.saleor.cloud/graphql/",
+        jwks: mockJwksValue,
+        appId: mockAppId,
+      };
+
+      await handler(req, res);
+
+      expect(mockOnAuthAplSaved).not.toHaveBeenCalled();
+      expect(mockOnAuthAplFailed).toHaveBeenCalledWith(
+        expect.anything(/* Assume original request */),
+        expect.objectContaining({
+          authData: expectedAuthData,
+          error: expect.objectContaining({
+            message: "test error",
+          }),
+        })
+      );
+    });
+
+    it("Allows to send custom error response via callback hook", async () => {
+      const mockOnRequestStart = vi.fn().mockImplementation((req, context: any) => {
```

Comment:
```suggestion
      const mockOnRequestStart = vi.fn().mockImplementation((req, context: unknown) => {
```
Or add ignore the comment. Since it's test code IMO it's allowed.